### ISSUE TITLE
Tray icon does not exist on GNOME Wayland

### ISSUE THREAD

**Issue by Commenter25** [hhannine/superpaper#126]
When using the default Wayland mode in GNOME, the tray icon will not work, even with [AppIndicator and KStatusNotifierItem Support](https://extensions.gnome.org/extension/615/appindicator-support/) installed. However, if running in Xorg mode, or if Superpaper is launched with the prefix ` GDK_BACKEND=x11`, the tray icon appears as expected (though it's very wide for some reason).

Tested on Fedora 36 + GNOME 42, with manually installed commit [0c502d3](https://github.com/hhannine/superpaper/commit/0c502d3532b5f230300f5fd00cf72c3003924850).

**Comment by hhannine** [comment]
Hmmh probably some problem of Wxpython and Wayland. Which installation type were you using? Thanks for the workaround. I don't know if I can do anything for this other than hope that an update for wxpython fixes this.

**Comment by KarkanAlzwayed** [comment]
I almost opened an issue about the same issue for KDE, but I think it is a similar issue? If not, please let me know, so I can start a new one. The icon doesn't show in the systray on wayland.

**Comment by jonathanjsimon** [comment]
@KarkanAlzwayed I am also on Plasma Wayland and the icon does not show for me either. Setting `GDK_BACKEND=x11` does resolve the issue.



```
❯ superpaper
/usr/lib/python3.11/site-packages
/home/jsimon/.config/superpaper
/home/jsimon/.config/superpaper/profiles

(superpaper:75735): Gtk-CRITICAL **: 08:27:13.935: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
Gtk-Message: 08:27:14.144: Failed to load module "appmenu-gtk-module"
^C
```

vs...

```
❯ GDK_BACKEND=x11 superpaper
/usr/lib/python3.11/site-packages
/home/jsimon/.config/superpaper
/home/jsimon/.config/superpaper/profiles
Gtk-Message: 08:27:18.030: Failed to load module "appmenu-gtk-module"
```

**Comment by KarkanAlzwayed** [comment]
> @KarkanAlzwayed I am also on Plasma Wayland and the icon does not show for me either. Setting `GDK_BACKEND=x11` does resolve the issue.
> 
> ```
> ❯ superpaper
> /usr/lib/python3.11/site-packages
> /home/jsimon/.config/superpaper
> /home/jsimon/.config/superpaper/profiles
> 
> (superpaper:75735): Gtk-CRITICAL **: 08:27:13.935: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
> Gtk-Message: 08:27:14.144: Failed to load module "appmenu-gtk-module"
> ^C
> ```
> 
> vs...
> 
> ```
> ❯ GDK_BACKEND=x11 superpaper
> /usr/lib/python3.11/site-packages
> /home/jsimon/.config/superpaper
> /home/jsimon/.config/superpaper/profiles
> Gtk-Message: 08:27:18.030: Failed to load module "appmenu-gtk-module"
> ```

Oh nice. Where do you put that line? In /etc/environment?

**Comment by jonathanjsimon** [comment]
> > @KarkanAlzwayed I am also on Plasma Wayland and the icon does not show for me either. Setting `GDK_BACKEND=x11` does resolve the issue.
> > ```
> > ❯ superpaper
> > /usr/lib/python3.11/site-packages
> > /home/jsimon/.config/superpaper
> > /home/jsimon/.config/superpaper/profiles
> > 
> > (superpaper:75735): Gtk-CRITICAL **: 08:27:13.935: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
> > Gtk-Message: 08:27:14.144: Failed to load module "appmenu-gtk-module"
> > ^C
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > vs...
> > ```
> > ❯ GDK_BACKEND=x11 superpaper
> > /usr/lib/python3.11/site-packages
> > /home/jsimon/.config/superpaper
> > /home/jsimon/.config/superpaper/profiles
> > Gtk-Message: 08:27:18.030: Failed to load module "appmenu-gtk-module"
> > ```
> 
> Oh nice. Where do you put that line? In /etc/environment?

Just put `GDK_BACKEND=x11` before `superpaper` in a shell when you launch it. If you modify the `.desktop` file, just stick that same value at the front of the `Exec=` entry before `/usr/bin/python3 -m superpaper`. Or wait for it to be fixed upstream.

**Comment by KarkanAlzwayed** [comment]
> > > @KarkanAlzwayed I am also on Plasma Wayland and the icon does not show for me either. Setting `GDK_BACKEND=x11` does resolve the issue.
> > > ```
> > > ❯ superpaper
> > > /usr/lib/python3.11/site-packages
> > > /home/jsimon/.config/superpaper
> > > /home/jsimon/.config/superpaper/profiles
> > > 
> > > (superpaper:75735): Gtk-CRITICAL **: 08:27:13.935: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
> > > Gtk-Message: 08:27:14.144: Failed to load module "appmenu-gtk-module"
> > > ^C
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > vs...
> > > ```
> > > ❯ GDK_BACKEND=x11 superpaper
> > > /usr/lib/python3.11/site-packages
> > > /home/jsimon/.config/superpaper
> > > /home/jsimon/.config/superpaper/profiles
> > > Gtk-Message: 08:27:18.030: Failed to load module "appmenu-gtk-module"
> > > ```
> > 
> > 
> > Oh nice. Where do you put that line? In /etc/environment?
> 
> Just put `GDK_BACKEND=x11` before `superpaper` in a shell when you launch it. If you modify the `.desktop` file, just stick that same value at the front of the `Exec=` entry before `/usr/bin/python3 -m superpaper`. Or wait for it to be fixed upstream.

I put that line in several locations in that exec line, and it is still nothing. Could you please copy how you have it in your .desktop and paste it here? Thank you

**Comment by jonathanjsimon** [comment]
> > > > @KarkanAlzwayed I am also on Plasma Wayland and the icon does not show for me either. Setting `GDK_BACKEND=x11` does resolve the issue.
> > > > ```
> > > > ❯ superpaper
> > > > /usr/lib/python3.11/site-packages
> > > > /home/jsimon/.config/superpaper
> > > > /home/jsimon/.config/superpaper/profiles
> > > > 
> > > > (superpaper:75735): Gtk-CRITICAL **: 08:27:13.935: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
> > > > Gtk-Message: 08:27:14.144: Failed to load module "appmenu-gtk-module"
> > > > ^C
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > vs...
> > > > ```
> > > > ❯ GDK_BACKEND=x11 superpaper
> > > > /usr/lib/python3.11/site-packages
> > > > /home/jsimon/.config/superpaper
> > > > /home/jsimon/.config/superpaper/profiles
> > > > Gtk-Message: 08:27:18.030: Failed to load module "appmenu-gtk-module"
> > > > ```
> > > 
> > > 
> > > Oh nice. Where do you put that line? In /etc/environment?
> > 
> > 
> > Just put `GDK_BACKEND=x11` before `superpaper` in a shell when you launch it. If you modify the `.desktop` file, just stick that same value at the front of the `Exec=` entry before `/usr/bin/python3 -m superpaper`. Or wait for it to be fixed upstream.
> 
> I put that line in several locations in that exec line, and it is still nothing. Could you please copy how you have it in your .desktop and paste it here? Thank you
> 
> Also, launching it through the terminal like you suggested means I have to keep the terminal open, otherwise it closes the icon whenever I close the terminal. Tried adding -d to the line for "detach" and it still closes the app when I closed the terminal

My bad, use `env GDK_BACKEND=x11` as a prefix rather than just `GDK_BACKEND=x11`.

```
❯ cat /usr/share/applications/superpaper.desktop
[Desktop Entry]
Type=Application
Version=1.0
Name=Superpaper
GenericName=Multi-monitor wallpaper manager
Exec=env GDK_BACKEND=x11 /usr/bin/python3 -m superpaper
Icon=superpaper
Terminal=false
Categories=Utility;
X-KDE-autostart-after=panel
```

If you have `superpaper` configured to start at login via Plasma's autostart settings pane, you'll also want to modify `/home/${USER}/.config/autostart/superpaper.desktop` with the same `env GDK_BACKEND=x11` prefix in the `Exec` entry:

````
[Desktop Entry]
Categories=Utility;
Exec=env GDK_BACKEND=x11 /usr/bin/python3 -m superpaper
GenericName=Multi-monitor wallpaper manager
Icon=superpaper
Name=Superpaper
Terminal=false
Type=Application
Version=1.0
X-KDE-autostart-after=panel
```

I'm sharing both because they are slightly different.

**Comment by KarkanAlzwayed** [comment]
> > > > > @KarkanAlzwayed I am also on Plasma Wayland and the icon does not show for me either. Setting `GDK_BACKEND=x11` does resolve the issue.
> > > > > ```
> > > > > ❯ superpaper
> > > > > /usr/lib/python3.11/site-packages
> > > > > /home/jsimon/.config/superpaper
> > > > > /home/jsimon/.config/superpaper/profiles
> > > > > 
> > > > > (superpaper:75735): Gtk-CRITICAL **: 08:27:13.935: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
> > > > > Gtk-Message: 08:27:14.144: Failed to load module "appmenu-gtk-module"
> > > > > ^C
> > > > > ```
> > > > > 
> > > > > 
> > > > >     
> > > > >       
> > > > >     
> > > > > 
> > > > >       
> > > > >     
> > > > > 
> > > > >     
> > > > >   
> > > > > vs...
> > > > > ```
> > > > > ❯ GDK_BACKEND=x11 superpaper
> > > > > /usr/lib/python3.11/site-packages
> > > > > /home/jsimon/.config/superpaper
> > > > > /home/jsimon/.config/superpaper/profiles
> > > > > Gtk-Message: 08:27:18.030: Failed to load module "appmenu-gtk-module"
> > > > > ```
> > > > 
> > > > 
> > > > Oh nice. Where do you put that line? In /etc/environment?
> > > 
> > > 
> > > Just put `GDK_BACKEND=x11` before `superpaper` in a shell when you launch it. If you modify the `.desktop` file, just stick that same value at the front of the `Exec=` entry before `/usr/bin/python3 -m superpaper`. Or wait for it to be fixed upstream.
> > 
> > 
> > I put that line in several locations in that exec line, and it is still nothing. Could you please copy how you have it in your .desktop and paste it here? Thank you
> > Also, launching it through the terminal like you suggested means I have to keep the terminal open, otherwise it closes the icon whenever I close the terminal. Tried adding -d to the line for "detach" and it still closes the app when I closed the terminal
> 
> My bad, use `env GDK_BACKEND=x11` as a prefix rather than just `GDK_BACKEND=x11`.
> 
> ```
> ❯ cat /usr/share/applications/superpaper.desktop
> [Desktop Entry]
> Type=Application
> Version=1.0
> Name=Superpaper
> GenericName=Multi-monitor wallpaper manager
> Exec=env GDK_BACKEND=x11 /usr/bin/python3 -m superpaper
> Icon=superpaper
> Terminal=false
> Categories=Utility;
> X-KDE-autostart-after=panel
> ```
> 
> If you have `superpaper` configured to start at login via Plasma's autostart settings pane, you'll also want to modify `/home/${USER}/.config/autostart/superpaper.desktop` with the same `env GDK_BACKEND=x11` prefix in the `Exec` entry:
> 
> ```
> [Desktop Entry]
> Categories=Utility;
> Exec=env GDK_BACKEND=x11 /usr/bin/python3 -m superpaper
> GenericName=Multi-monitor wallpaper manager
> Icon=superpaper
> Name=Superpaper
> Terminal=false
> Type=Application
> Version=1.0
> X-KDE-autostart-after=panel
> ```
> 
> I'm sharing both because they are slightly different.

I copied both from yours directly and it doesn't fully work. Clicking on the app doesn't launch it or even put the icon in the panel. If I reboot the machine, the app is already in the panel and functions no problem. The if I right click the icon and exit it, then try to launch it again, nope. It will never launch until I reboot again. So, I think the second one is working, but the first one isn't for some reason. See this is why I stay away from wayland, it is these little things that frustrate me to hell and back.

**Comment by jonathanjsimon** [comment]
Yep, I'm only checking out Wayland occasionally and early because Plasma 6 won't have X11 as an option (or so I've heard around the internet) and I want to iron out these types of issues while I still have X11 as a fallback.

Not sure what is preventing your global `.desktop` file from operating....

**Comment by KarkanAlzwayed** [comment]
> Yep, I'm only checking out Wayland occasionally and early because Plasma 6 won't have X11 as an option (or so I've heard around the internet) and I want to iron out these types of issues while I still have X11 as a fallback.
> 
> Not sure what is preventing your global `.desktop` file from operating....

Only Fedora is dropping x11, and you can still re-install it if you wanted to. I use endeavour OS and we are good over here. Off I go back to X11. Not gonna fuss over this anymore. X11 is working just fine for me. I appreciate your help

**Comment by Cris70** [comment]
Any news on this bug?

**Comment by Turkeysteaks** [comment]
Just want to update that this issue is still present on Plasma 6 KDE on Arch with the AUR version at least.
The issue is indeed fixed by adding `GDK_BACKEND=x11` as either an ENV in plasma's autostart or just by setting `alias superpaper = 'GDK_BACKEND=x11 superpaper'` in `.bashrc`

**Comment by kalzEOS** [comment]
> Just want to update that this issue is still present on Plasma 6 KDE on Arch with the AUR version at least. The issue is indeed fixed by adding `GDK_BACKEND=x11` as either an ENV in plasma's autostart or just by setting `alias superpaper = 'GDK_BACKEND=x11 superpaper'` in `.bashrc`

That worked. Thank you so much. I copied the app into` ./local/share/applications` so it's for the user only and won't be overidden by updates. and made my changes there. Changed the `Exec` line to this `Exec=env GDK_BACKEND=x11 python3 -m superpaper ` in the `.desktop` file. Added this line `alias superpaper env GDK_BACKEND=x11 superpaper` to my `fish.config` since I use fish shell. I even changed the app icon and the system tray icon to something colorful instead of the one the dev has. 😅

**Comment by Ohp3ueyi** [comment]
> Just want to update that this issue is still present on Plasma 6 KDE on Arch with the AUR version at least. The issue is indeed fixed by adding `GDK_BACKEND=x11` as either an ENV in plasma's autostart or just by setting `alias superpaper = 'GDK_BACKEND=x11 superpaper'` in `.bashrc`

Can you interact with the icon? i can see it in the system tray but clicking it does nothing except for asking for "remote control request" sometimes

**Comment by Cris70** [comment]
> Can you interact with the icon? i can see it in the system tray but clicking it does nothing except for asking for "remote control request" sometimes

I can, though usually not on the first right-click. But if I insist, I can usually get a context menu.
One (small) problem of using this work-around, at least in my case, is that the tray icon loses it's name, and instead I get `__main__.py` as name.

**Comment by kalzEOS** [comment]
> > Just want to update that this issue is still present on Plasma 6 KDE on Arch with the AUR version at least. The issue is indeed fixed by adding `GDK_BACKEND=x11` as either an ENV in plasma's autostart or just by setting `alias superpaper = 'GDK_BACKEND=x11 superpaper'` in `.bashrc`
> 
> Can you interact with the icon? i can see it in the system tray but clicking it does nothing except for asking for "remote control request" sometimes

Yes, it works no issue at all. I did get the remote access prompt and gave it permission. Checked both boxes (if you get them) and now it works no issues on my end

**Comment by Ohp3ueyi** [comment]
> > > Just want to update that this issue is still present on Plasma 6 KDE on Arch with the AUR version at least. The issue is indeed fixed by adding `GDK_BACKEND=x11` as either an ENV in plasma's autostart or just by setting `alias superpaper = 'GDK_BACKEND=x11 superpaper'` in `.bashrc`
> > 
> > 
> > Can you interact with the icon? i can see it in the system tray but clicking it does nothing except for asking for "remote control request" sometimes
> 
> Yes, it works no issue at all. I did get the remote access prompt and gave it permission. Checked both boxes (if you get them) and now it works no issues on my end

well i don't get checkboxes, i get only question if i want to give access to input devices and buttons to share or cancel. if i press rightclick, it seems to "hold" this action and opens context menu for first application i move mouse to (plasma or some other app in tray). --debug doesn't show anything on clicking. (AUR installation (BTW))

**Comment by kalzEOS** [comment]
> > > > Just want to update that this issue is still present on Plasma 6 KDE on Arch with the AUR version at least. The issue is indeed fixed by adding `GDK_BACKEND=x11` as either an ENV in plasma's autostart or just by setting `alias superpaper = 'GDK_BACKEND=x11 superpaper'` in `.bashrc`
> > > 
> > > 
> > > Can you interact with the icon? i can see it in the system tray but clicking it does nothing except for asking for "remote control request" sometimes
> > 
> > 
> > Yes, it works no issue at all. I did get the remote access prompt and gave it permission. Checked both boxes (if you get them) and now it works no issues on my end
> 
> well i don't get checkboxes, i get only question if i want to give access to input devices and buttons to share or cancel. if i press rightclick, it seems to "hold" this action and opens context menu for first application i move mouse to (plasma or some other app in tray). --debug doesn't show anything on clicking. (AUR installation (BTW))

I have the AUR package, too. Do you accept the remote access prompt or do you hit cancel? Make sure to accept it. If not, let me know and I'll post what I have exactly and see if it helps you fix it. Not home at at the moment

**Comment by Ohp3ueyi** [comment]
> > > > > Just want to update that this issue is still present on Plasma 6 KDE on Arch with the AUR version at least. The issue is indeed fixed by adding `GDK_BACKEND=x11` as either an ENV in plasma's autostart or just by setting `alias superpaper = 'GDK_BACKEND=x11 superpaper'` in `.bashrc`
> > > > 
> > > > 
> > > > Can you interact with the icon? i can see it in the system tray but clicking it does nothing except for asking for "remote control request" sometimes
> > > 
> > > 
> > > Yes, it works no issue at all. I did get the remote access prompt and gave it permission. Checked both boxes (if you get them) and now it works no issues on my end
> > 
> > well i don't get checkboxes, i get only question if i want to give access to input devices and buttons to share or cancel. if i press rightclick, it seems to "hold" this action and opens context menu for first application i move mouse to (plasma or some other app in tray). --debug doesn't show anything on clicking. (AUR installation (BTW))
> 
> I have the AUR package, too. Do you accept the remote access prompt or do you hit cancel? Make sure to accept it. If not, let me know and I'll post what I have exactly and see if it helps you fix it. Not home at at the moment 

Yes i accept it, of course. I tried also on a fresh user with the same result. It may be as well some issue with python on my system but I'll appreciate it if you can share what you did, I'll try to do it also on my other machine. 
It's not a huge problem for me, the program works and i can configure it on startup, also I'm using community made software for more than a quarter of a century so I'm not too demanding, but if we can make it work as intended it'll be great.

**Comment by kalzEOS** [comment]
> > > > > > Just want to update that this issue is still present on Plasma 6 KDE on Arch with the AUR version at least. The issue is indeed fixed by adding `GDK_BACKEND=x11` as either an ENV in plasma's autostart or just by setting `alias superpaper = 'GDK_BACKEND=x11 superpaper'` in `.bashrc`
> > > > > 
> > > > > 
> > > > > Can you interact with the icon? i can see it in the system tray but clicking it does nothing except for asking for "remote control request" sometimes
> > > > 
> > > > 
> > > > Yes, it works no issue at all. I did get the remote access prompt and gave it permission. Checked both boxes (if you get them) and now it works no issues on my end
> > > 
> > > 
> > > well i don't get checkboxes, i get only question if i want to give access to input devices and buttons to share or cancel. if i press rightclick, it seems to "hold" this action and opens context menu for first application i move mouse to (plasma or some other app in tray). --debug doesn't show anything on clicking. (AUR installation (BTW))
> > 
> > 
> > I have the AUR package, too. Do you accept the remote access prompt or do you hit cancel? Make sure to accept it. If not, let me know and I'll post what I have exactly and see if it helps you fix it. Not home at at the moment
> 
> Yes i accept it, of course. I tried also on a fresh user with the same result. It may be as well some issue with python on my system but I'll appreciate it if you can share what you did, I'll try to do it also on my other machine. It's not a huge problem for me, the program works and i can configure it on startup, also I'm using community made software for more than a quarter of a century so I'm not too demanding, but if we can make it work as intended it'll be great.

My apologies for taking so long to answer. Had so many things to get done. 
Here is what I did : 
Pre-reqs just for a general knowledge.
If you installed the app from the AUR, then it should be in 3 places:
1. The launch script in `/usr/bin/superpaper` -- this is just a small python script
2. The .dekstop file `superpaper.desktop` that is in `/usr/share/applications` -- this is what creates the desktop app and shows in the panel when you launch it
3. The actual app and all of its components that is in here `/usr/lib/python3.X/site-packages/superpaper/` - in this one, you can even change how the app functions. You can change the system tray icon if you don't like it, like I did (optional)

The process:
1. I copied the .desktop file from `/usr/share/applications` (system wide) to `~/.local/share/applications` (current user only and won't be messed up by any updates you get for the app)
2. Edited the .desktop file in `~/.local/share/applications` changed the `Exec` line. Here is what mine has 
```
[Desktop Entry]
Categories=Utility;
Comment=
#Exec=env GDK_BACKEND=x11 python3 -m superpaper #use this line if the one under doesn't work for some reason
Exec=env GDK_BACKEND=x11 superpaper
GenericName=Multi-monitor wallpaper manager
Icon=/path/to your/favoriteicon.png #this is optional. I just didn't like the default icon
Name=Superpaper
NoDisplay=false
Path=
StartupNotify=true
Terminal=false
TerminalOptions=
Type=Application
Version=1.0
X-KDE-SubstituteUID=false
X-KDE-Username=
X-KDE-autostart-after=panel
```
You are done here and it should work. But you should add this line `alias superpaper='env GDK_BACKEND=x11 superpaper'`
 to the end of your `.bashrc` file, that way you can run the app from terminal to troubleshoot it in case it doesn't work for you. 
- Make sure you have the following installed on your system before doing any of the above:
1. Python
2. python-pillow
3. python-pyqt5
4. python-setuptools

**Optional**
Change the system tray icon to match the one you changed it to in the .desktop file. You do that by going into `/usr/lib/python3.13/site-packages/superpaper/resources/` and replace the `superpaper.png` icon that is there with your own (or the one I have provided below). Make sure you name it exactly as the old one, `superpaper.png`. Either remove or rename the old one by appending `.bak` to the end of it. This should do it.

Here is the one I used for both

![Image](https://github.com/user-attachments/assets/c830b508-f2cf-4aeb-a695-2c2d69a24d32)

**Comment by Ohp3ueyi** [comment]
> My apologies for taking so long to answer. Had so many things to get done. Here is what I did : Pre-reqs just for a general knowledge. If you installed the app from the AUR, then it should be in 3 places:
> 
> 1. The launch script in `/usr/bin/superpaper` -- this is just a small python script
> 2. The .dekstop file `superpaper.desktop` that is in `/usr/share/applications` -- this is what creates the desktop app and shows in the panel when you launch it
> 3. The actual app and all of its components that is in here `/usr/lib/python3.X/site-packages/superpaper/` - in this one, you can even change how the app functions. You can change the system tray icon if you don't like it, like I did (optional)
> 
> The process:
> 
> 1. I copied the .desktop file from `/usr/share/applications` (system wide) to `~/.local/share/applications` (current user only and won't be messed up by any updates you get for the app)
> 2. Edited the .desktop file in `~/.local/share/applications` changed the `Exec` line. Here is what mine has
> 
> ```
> [Desktop Entry]
> Categories=Utility;
> Comment=
> #Exec=env GDK_BACKEND=x11 python3 -m superpaper #use this line if the one under doesn't work for some reason
> Exec=env GDK_BACKEND=x11 superpaper
> GenericName=Multi-monitor wallpaper manager
> Icon=/path/to your/favoriteicon.png #this is optional. I just didn't like the default icon
> Name=Superpaper
> NoDisplay=false
> Path=
> StartupNotify=true
> Terminal=false
> TerminalOptions=
> Type=Application
> Version=1.0
> X-KDE-SubstituteUID=false
> X-KDE-Username=
> X-KDE-autostart-after=panel
> ```
> 
> You are done here and it should work. But you should add this line `alias superpaper='env GDK_BACKEND=x11 superpaper'` to the end of your `.bashrc` file, that way you can run the app from terminal to troubleshoot it in case it doesn't work for you.
> 
> * Make sure you have the following installed on your system before doing any of the above:
> 
> 1. Python
> 2. python-pillow
> 3. python-pyqt5
> 4. python-setuptools
> 
> **Optional** Change the system tray icon to match the one you changed it to in the .desktop file. You do that by going into `/usr/lib/python3.13/site-packages/superpaper/resources/` and replace the `superpaper.png` icon that is there with your own (or the one I have provided below). Make sure you name it exactly as the old one, `superpaper.png`. Either remove or rename the old one by appending `.bak` to the end of it. This should do it.
> 
> Here is the one I used for both
> 
> ![Image](https://github.com/user-attachments/assets/c830b508-f2cf-4aeb-a695-2c2d69a24d32)

It didn't take long at all, thank You. So it seems like problem is with my system. Apparently on my other computer (more less the same installation, also Arch (btw of course), also AUR and more-less the same user settings) it's working fine - i can right click. On the other hand on different user on main machine i can't right click, so this leads me to conclusion that it's some issue with python, not superpaper itself. I rebuilt all python packages but no success.
As i pointed, it's not a big problem for me, so if it affects only my installation let's treat it as just a minor inconvenience - i can live with that, my wallpapers spread beautifully across monitors anyway.  There will always be some misunderstandings between qt versions, window system protocols and python versions. I can fully accept that.
Best regards and thanks.
